### PR TITLE
Add JDK 17 support

### PR DESCRIPTION
**Pull request by dai-chen**
### Description

1. Add JDK 17 to CI/CD. To pass all SQL build, `lombok`, `mockito` and `jacoco` is upgraded to newer version accordingly. 
2. Add JDK 17 to SQL JDBC CI/CD as well.
3. Update developer guide doc to clarify the JDK version supported.

### Issues Resolved
https://github.com/opensearch-project/sql/issues/467
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/sql-jdbc-test-and-build-workflow.yml [hunk 1]
```diff
@@ -4,19 +4,23 @@ on: [push, pull_request]
 
 jobs:
   build:
-
+    strategy:
+      matrix:
+        java:
+          - 14
+          - 17
```
Comment (vamsi-amazon) [inline .github/workflows/sql-jdbc-test-and-build-workflow.yml:11]: Should we include 11 as well?

Comment (dai-chen) [reply .github/workflows/sql-jdbc-test-and-build-workflow.yml:12]: done. thanks!

**Review by penghuo — APPROVED** [review]
LGTM

**Review by vamsi-amazon — APPROVED** [review]
LGTM

**Pull request merged**